spring-batch-docs/asciidoc/spring-batch-integration.adoc

```diff
@@ -800,73 +767,62 @@ throttle limits and so on, when implementing your use case.
-public JmsSendingMessageHandler jmsOutboundChannelAdapter() {
-	JmsSendingMessageHandler jmsSendingMessageHandler = new JmsSendingMessageHandler(jmsTemplate());
-	jmsSendingMessageHandler.setDestinationName("requests");
-	return jmsSendingMessageHandler;
+    MessagingTemplate template = new MessagingTemplate();
+    template.setDefaultChannel(requests());
+    template.setReceiveTimeout(2000);
+    return template;
 }
 
 @Bean
 @StepScope()
 public ChunkMessageChannelItemWriter itemWriter() {
-	ChunkMessageChannelItemWriter chunkMessageChannelItemWriter = new ChunkMessageChannelItemWriter();
-	chunkMessageChannelItemWriter.setMessagingOperations(messagingTemplate());
-	chunkMessageChannelItemWriter.setReplyChannel(replies());
-	return chunkMessageChannelItemWriter;
+    ChunkMessageChannelItemWriter chunkMessageChannelItemWriter = new ChunkMessageChannelItemWriter();
+    chunkMessageChannelItemWriter.setMessagingOperations(messagingTemplate());
+    chunkMessageChannelItemWriter.setReplyChannel(replies());
+    return chunkMessageChannelItemWriter;
 }
 
 @Bean
 public RemoteChunkHandlerFactoryBean chunkHandler() {
-	RemoteChunkHandlerFactoryBean remoteChunkHandlerFactoryBean = new RemoteChunkHandlerFactoryBean();
-	remoteChunkHandlerFactoryBean.setChunkWriter(itemWriter());
-	remoteChunkHandlerFactoryBean.setStep(step1());
-	return remoteChunkHandlerFactoryBean;
+    RemoteChunkHandlerFactoryBean remoteChunkHandlerFactoryBean = new RemoteChunkHandlerFactoryBean();
+    remoteChunkHandlerFactoryBean.setChunkWriter(itemWriter());
+    remoteChunkHandlerFactoryBean.setStep(step1());
+    return remoteChunkHandlerFactoryBean;
 }
 
 @Bean
-public DefaultMessageListenerContainer defaultMessageListenerContainer() {
-	DefaultMessageListenerContainer defaultMessageListenerContainer = new DefaultMessageListenerContainer();
-	defaultMessageListenerContainer.setDestinationName("replies");
-	return defaultMessageListenerContainer;
-}
-
-@Bean
-public ChannelPublishingJmsMessageListener channelPublishingJmsMessageListener() {
-	ChannelPublishingJmsMessageListener channelPublishingJmsMessageListener = new  ChannelPublishingJmsMessageListener();
-	return channelPublishingJmsMessageListener;
+public QueueChannel replies() {
+    return new QueueChannel();
 }
 
 @Bean
-public JmsMessageDrivenEndpoint jmsReplies() {
-	JmsMessageDrivenEndpoint jmsMessageDrivenEndpoint = new JmsMessageDrivenEndpoint(defaultMessageListenerContainer(), channelPublishingJmsMessageListener());
-	jmsMessageDrivenEndpoint.setOutputChannel(replies());
-	return jmsMessageDrivenEndpoint;
+public IntegrationFlow jmsReplies() {
+    return IntegrationFlows.from(Jms.inboundAdapter(connectionFactory())
+                    .configureJmsTemplate(t ->
+                            t.deliveryPersistent(true).jmsMessageConverter(new SimpleMessageConverter()))
+                    .destination("replies"), c -> c.poller(Pollers.fixedRate(1000).maxMessagesPerPoll(1)))
```

Comment:
`Jms.messageDrivenChannelAdapter()` will be better and definitely will reflect the previous XML variant.